dev-tex/rail/files/rail-1.2_p1-C23.patch

```diff
@@ -0,0 +1,380 @@
+ 
+-freerule(rule)
+-RULETYPE *rule;
++void freerule(RULETYPE *rule)
+ {
+ 	RULETYPE *rulep;
+ 
+@@ -591,8 +567,7 @@
+ 
+ /* add two rule lists */
+ 
+-RULETYPE *addrule(rule1,rule2)
+-RULETYPE *rule1, *rule2;
++RULETYPE *addrule(RULETYPE *rule1, RULETYPE *rule2)
+ {
+ 	RULETYPE *rulep;
+ 
+@@ -609,8 +584,7 @@
+ 
+ /* output a rule list */
+ 
+-outrule(rule)
+-RULETYPE *rule;
++void outrule(RULETYPE *rule)
+ {
+ 	while(rule!=NULL) {
+ 
+@@ -625,8 +599,7 @@
+ 
+ /* look up an identifier */
+ 
+-IDTYPE *lookup(name)
+-char *name;
++IDTYPE *lookup(char *name)
+ {
+ 	IDTYPE *idp, **idq;
+ 
+@@ -648,8 +621,7 @@
+ 
+ /* delete an identifier */
+ 
+-delete(id)
+-IDTYPE *id;
++void delete(IDTYPE *id)
+ {
+ 	IDTYPE *idp, **idq;
+ 
+@@ -665,7 +637,7 @@
+ 
+ /* check that there are no undefined identifiers */
+ 
+-checkdefs()
++void checkdefs(void)
+ {
+ 	IDTYPE *id;
+ 
+@@ -676,8 +648,7 @@
+ 
+ /* complain about an undefined identifier */
+ 
+-undef(id)
+-IDTYPE *id;
++void undef(IDTYPE *id)
+ {
+ 	if(chkgram)
+ 		error("undefined identifier '%s'",id->name);
+@@ -685,8 +656,7 @@
+ 
+ /* complain about a redefined identifier */
+ 
+-redef(id)
+-IDTYPE *id;
++void redef(IDTYPE *id)
+ {
+ 	if(chkgram)
+ 		error("redefined identifier '%s'",id->name);
+@@ -694,8 +664,7 @@
+ 
+ /* display an error */
+ 
+-error(f,s)
+-char *f, *s;
++void error(char *f, char *s)
+ {
+ 	if(newline) {
+ 		printf("\n");
+@@ -713,8 +682,7 @@
+ 	fprintf(stderr,"\n");
+ }
+ 
+-fatal(f,s)
+-char *f,*s;
++void fatal(char *f, char *s)
+ {
+ 	error(f,s);
+ 
+--- a/rail.h
++++ b/rail.h
+@@ -68,22 +68,31 @@
+ 
+ extern IDTYPE *errorid;
+ 
+-extern char *mcheck();
++extern char *mcheck(char *s);
+ 
+-extern BODYTYPE *newbody();
+-extern freebody();
+-extern int isemptybody();
+-extern BODYTYPE *addbody();
+-extern BODYTYPE *revbody();
+-
+-extern RULETYPE *newrule();
+-extern freerule();
+-extern RULETYPE *addrule();
+-extern outrule();
+-
+-extern IDTYPE *lookup();
+-extern delete();
+-
+-extern undef();
+-extern redef();
+-extern error();
++extern BODYTYPE *newbody(int kind, BODYTYPE *body1, BODYTYPE *body2);
++extern void freebody(BODYTYPE *body);
++extern int isemptybody(BODYTYPE *body);
++extern BODYTYPE *addbody(int kind, BODYTYPE *body1, BODYTYPE *body2);
++extern BODYTYPE *revbody(BODYTYPE *body);
++extern void posbody(BODYTYPE *body, int ystart);
++extern void fmtbody(BODYTYPE *body, char *cent, char arrow);
++
++extern RULETYPE *newrule(IDTYPE *id, BODYTYPE *body);
++extern void freerule(RULETYPE *rule);
++extern RULETYPE *addrule(RULETYPE *rule1, RULETYPE *rule2);
++extern void outrule(RULETYPE *rule);
++
++extern IDTYPE *lookup(char *name);
++extern void delete(IDTYPE *id);
++
++extern void undef(IDTYPE *id);
++extern void redef(IDTYPE *id);
++extern void error(char *f, char *s);
++extern void yyerror(char *s);
++extern int yywrap(void);
++
++int setopt(char c, char *s);
++void usage(void);
++void checkdefs(void);
++void fatal(char *f, char *s);
+--- a/Makefile
++++ b/Makefile
```

Comment:
I think this should be in a separate patch file. It shouldn't be mixed in with the C23 changes.